### PR TITLE
Email base proposal comment

### PR DESCRIPTION
**Pull request by vigneshsarma**
`Emailer`: minimal infrastructure for sending out email
`Proposal service`: adds `send_email` for `proposal_comment`.
`Templates`: These are very basic.

BugFix: 445e5d1 fix the permission issue, that prevents normal but authenticated user from commenting

Fix for #7,  #10 
#### TODO: 
- Derivation of correct host is very ugly. We need a better way to handle this.
- Once we add a delayed job framework like celery or rq we should make this a delayed task.

### CODE REVIEW

File: junction/emailer.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from django.conf import settings
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+
+
+def send_email(to, context, template_dir):
+    """Render given templates and send email to `to`.
+
+    :param to: User object to send email to..
+    :param context:
```
Comment (kracekumar) [inline junction/emailer.py]: Add a line saying `dict containing which needs to be passed to django template` ?

File: junction/proposals/views.py [hunk 2]
```diff
@@ -193,18 +194,20 @@ def create_proposal_comment(request, conference_slug, proposal_slug):
...
-                                       )
+        if private and not _is_proposal_author_or_reviewer(
+                request.user, conference, proposal):
+            raise Http404()
+
+        proposal_comment = ProposalComment.objects.create(
+            proposal=proposal, comment=comment,
+            private=private, commenter=request.user)
+        send_mail_for_new_comment(proposal_comment,
+                                  host='http://%s' % request.META['HTTP_HOST'])
```
Comment (kracekumar) [inline junction/proposals/views.py]: I think we shouldn't hard code `http`, how about reading it from `SITE_PROTOCOL`.

File: junction/templates/proposals/email/message.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+There is a new comment on "<a href="{{host}}{{proposal.get_absolute_url}}">
+    {{proposal.title}}
+</a>" by <b>{{comment.commenter}}</b>:
+<br/>
+"{{comment.comment}}"
+
+<hr/>
```
Comment (kracekumar) [inline junction/templates/proposals/email/message.html]: Add "Please, don't reply to the email. You can login to <url> and leave a comment"

**Comment by kracekumar** [comment]
Every thing else looks ok. Merge it once done.

File: junction/proposals/views.py (hunk 2 above)
Comment (kracekumar) [inline junction/proposals/views.py]: Sorry, didn't read the `TODO` section, please create a bug or fix this in the pull request.

**Comment by kracekumar** [comment]
It would be nice to see email screen shoot of new comment.

**Comment by vigneshsarma** [comment]
Screen:
In gmail

**Comment by kracekumar** [comment]
How about adding login url with next query string ?

**Pull request merged**